### ISSUE TITLE
Анализ графа задач

### ISSUE THREAD

**Issue by mmuddy** [mmuddy/taskgraph#7]
Необходимо реализовать анализ графа:
- поиск полного путь от задачи А (или набора задач) к задаче Б (или набору задач),
- разделение полного пути от одного набора задач к другому набору задач на отдельные, максимально параллельные Н путей (распределение задач между Н участниками),
- выявление задач, являющихся самыми сильными зависимостями (на графе выглядят как задача с большим исходящим пучком стрелок),
- выявление отдельных задач в полном пути, которые хорошо бы декомпозировать/переформулировать/устранить зависимости (таких, которые мешают параллельному исполнению, например как задачи из пункта выше).

**Comment by mmuddy** [comment]
Не ясно о чем идет речь. Требуются дополнительные пояснения.

**Comment by zmm** [comment]
Речь здесь идет примерно о следующем. Допустим, у вас есть некоторый граф выполнения задач. Вам необходимо по нему определить:
- полный путь от задачи А (или набора задач) к задаче Б (или набору задач) 
- разделение полного пути от одного набора задач к другому набору задач на отдельные, максимально параллельные Н путей (распределение задач между Н участниками)
- какие задачи являются самыми сильными зависимостями (на графе выглядят как задача с большим исходящим пучком стрелок)
- маркировка отдельных задач в полном пути, которые хорошо бы декомпозировать/перефомулировать/устранить зависимости (таких, которые мешают параллельному исполнению, например как задачи из пункта выше)

**Comment by mmuddy** [comment]
> какие задачи являются самыми сильными зависимостями (на графе выглядят как задача с большим исходящим пучком стрелок)

Как формально определить задачу с сильными зависимостями?
Что то типо если степень вершины больше эн сигм, то вершина является сильной зависимостью?

**Comment by zmm** [comment]
Да, давайте считать, что это такая вершина, где выходит более N стрелок. Хорошо бы N при этом вынести в настройки.
